server/routes/users.js

```diff
@@ -4,17 +4,26 @@ const router = express.Router()
 const { User } = require( '../database' )
 
 router.post( '/sign-up', ( request, response ) => {
+  //if request.session.user, don't even let someone see this page
   const { username, email, password, lat, long } = request.body
 
   User.signUp({ username, email, password, lat, long })
     .then( user => response.status( 201 ).json( user ))
     .catch( error => response.status( 500 ).json({ error: "The email or username you entered were already taken." }))
+    //TODO: add handling (maybe on front end?) if there is missing info. that creates an error
 })
 
 router.post( '/login', ( request, response ) => {
+  //if request.session.user, don't even let someone see this page
```

Comment:
Yes. Add a middleware function that checks this.

Reply:
ok